src/passes/Asyncify.cpp

```diff
@@ -298,47 +298,68 @@ enum class DataOffset { BStackPos = 0, BStackEnd = 4 };
 
 const auto STACK_ALIGN = 4;
 
-// A helper class for managing fake global names. Creates the globals and
+// A helper class for managing fake call names. Creates the targets and
 // provides mappings for using them.
-class GlobalHelper {
+// Fake calls are used to stash and then use return values from calls. We
+// need to store them somewhere that is valid Binaryen IR, but also will be
+// ignored by the Asyncify instrumentation, so we don't want to use a local.
+// What we do is replace the local used to receive a call's result with a
+// fake call to stash it, then do a fake call to receive it afterwards. (We
+// do it in two steps so that if we are async, we only do the first and
+// not the second, i.e., we don't store to the target local if not running
+// normally).
+class FakeCallHelper {
   Module& module;
 
 public:
-  GlobalHelper(Module& module) : module(module) {
-    map[Type::i32] = "asyncify_fake_call_global_i32";
-    map[Type::i64] = "asyncify_fake_call_global_i64";
-    map[Type::f32] = "asyncify_fake_call_global_f32";
-    map[Type::f64] = "asyncify_fake_call_global_f64";
+  FakeCallHelper(Module& module) : module(module) {
     Builder builder(module);
-    for (auto& pair : map) {
-      auto type = pair.first;
-      auto name = pair.second;
-      rev[name] = type;
-      module.addGlobal(builder.makeGlobal(
-        name, type, LiteralUtils::makeZero(type, module), Builder::Mutable));
+    std::string prefix = "__asyncify_fake_call";
+    for (auto type : {Type::i32, Type::i64, Type::f32, Type::f64}) {
+      auto typedPrefix = prefix + '_' + Type(type).toString() + '_';
```

Comment:
Nit: Can we move the final `'_'` to the "get" and "set" strings below? I think that would be slightly clearer and more consistent.

Reply:
Sure, changing.